### DISCUSSION TITLE
Help with setup

### DISCUSSION THREAD

**Discussion by kirilldem** [elden1337/hass-peaq#171] [General]
Hello,

Thanks for the addon, it sounds really useful!

During the setup, I have added my glow sensor that measures power usage in real time along with my status of the Easee charger (_sensor.carport_status_ - the one that says charging, ready_to_charge) as _Charger id_.

When I enable the on the peaq charge nothing happens. Am I missing something?

No nordpool pricing is selected or limited hours of operation. I am on version 2022.11.1 and 2.1.0 installed through HACS

**Comment by elden1337** [comment]
Hey, glad you found us! ;). 

It seems as if you ve added a faulty charger id. It will not be the sensor from HA but the s/n value written on your easee when it was new and now in your guide. I think about 8 characters long. 

Since this item is not an optional and the setup has failed you will need to uninstall - install peaqev. 

Best regards, Magnus

**Comment by kirilldem** [comment]
> Hey, glad you found us! ;).
> 
> It seems as if you ve added a faulty charger id. It will not be the sensor from HA but the s/n value written on your easee when it was new and now in your guide. I think about 8 characters long.
> 
> Since this item is not an optional and the setup has failed you will need to uninstall - install peaqev.
> 
> Best regards, Magnus

Thanks for the feedback and that makes sense :)

I removed the integration, restarted it and then went through the setup again. I added my serial number to Charge id. Then I turn on the peaqev charge, but the car still draws ~3.5kw. How long should it take to reduce the charge speed?
Am I missing something?

**Comment by elden1337** [comment]
1, when using easee people often have other snart charging active (ie Tibber). This must be turned off otherwise that will try to start it all the time. 

2. What does the sensors say? Does chargecontroller say "Start"? If so the charger is allowed to charge. 
What does the allowed current state?

**Reply by kirilldem** [reply]
> 

I took some screenshots - this is what it looks like before i plug in the charger:
![2022-11-04 16 49 38](https://user-images.githubusercontent.com/44521162/200026989-b7ed4321-c6dd-4270-b0ea-d12b54d0f0a3.png)
![2022-11-04 16 49 07](https://user-images.githubusercontent.com/44521162/200026992-e67d51bc-f0ee-4b11-ad4b-d4cb67654891.png)

And this is the charging status 5 minutes after. I do not have tibber either. 

![2022-11-04 16 54 11](https://user-images.githubusercontent.com/44521162/200027024-d6f89eda-1abd-42a2-b71c-d2e31e233c52.png)

**Reply by kirilldem** [reply]
Looking through the Wiki and it looks like I am missing some sensors:
sensor.peaqev_charger_controller 
sensor.peaqev_hour_controller
sensor.peaqev_threshold
You can see that I only have one sensor under peaqev. I installed it through HACS

**Reply by elden1337** [reply]
indeed you do. your logs probably say something about an issue. can you have a look there? [homeassistanturl]/config/logs

**Reply by elden1337** [reply]
you must have managed to miss something that i don't check for properly. so when we find what went wrong i can probably fix it for the next guy by validating more in setup

**Reply by kirilldem** [reply]
So there was a whole bunch of errors - I have both the nordpool and latest Easee integration mentioned in the docs. 

The nordpool integration has only one entity - nordpool_kwh_se3_sek_2_07_025

```
Logger: peaqevcore.models.hub.chargerobject
Source: custom_components/peaqev/peaqservice/hub/hub.py:51
Integration: peaqev ev-charging ([documentation](https://github.com/elden1337/hass-peaq), [issues](https://github.com/elden1337/hass-peaq/issues))
First occurred: 11:33:26 PM (1 occurrences)
Last logged: 11:33:26 PM

Chargerobject-state not found in given state-list. Value was: None
```

```
Logger: peaqevcore.models.hub.chargerswitch
Source: custom_components/peaqev/peaqservice/hub/hub.py:51
Integration: peaqev ev-charging (documentation, issues)
First occurred: 11:33:26 PM (1 occurrences)
Last logged: 11:33:26 PM

Chargerobject state was None while getting current. Updatecurrent will still process but you may see a missmatch in frontend.
```
```
Logger: homeassistant.components.sensor
Source: custom_components/peaqev/sensors/session_sensor.py:50
Integration: Sensor (documentation, issues)
First occurred: 11:33:26 PM (1 occurrences)
Last logged: 11:33:26 PM

Error while setting up peaqev platform for sensor
Traceback (most recent call last):
  File "/usr/src/homeassistant/homeassistant/helpers/entity_platform.py", line 281, in _async_setup_platform
    await asyncio.shield(task)
  File "/config/custom_components/peaqev/sensor.py", line 34, in async_setup_entry
    peaqsensors = await _helper.gather_sensors(hub, config)
  File "/config/custom_components/peaqev/sensors/create_sensor_helper.py", line 31, in gather_sensors
    ret.append(PeaqSessionCostSensor(hub, config.entry_id))
  File "/config/custom_components/peaqev/sensors/session_sensor.py", line 50, in __init__
    self._attr_unit_of_measurement = self._hub.nordpool.currency
AttributeError: 'HomeAssistantHub' object has no attribute 'nordpool'

```

```
This error originated from a custom integration.

Logger: custom_components.peaqev.peaqservice.hub.hub
Source: custom_components/peaqev/peaqservice/hub/hub.py:147
Integration: peaqev ev-charging (documentation, issues)
First occurred: 11:33:27 PM (187 occurrences)
Last logged: 11:35:18 PM

Unable to handle data: sensor.house_power_consumption ('HomeAssistantHub' object has no attribute 'nordpool') <state sensor.house_power_consumption=629; state_class=measurement, unit_of_measurement=W, device_class=power, icon=mdi:flash-outline, friendly_name=House - Power Consumption @ 2022-11-04T23:35:15.858878+01:00>|<state sensor.house_power_consumption=623; state_class=measurement, unit_of_measurement=W, device_class=power, icon=mdi:flash-outline, friendly_name=House - Power Consumption @ 2022-11-04T23:35:16.446685+01:00>
Unable to handle data: sensor.house_power_consumption ('HomeAssistantHub' object has no attribute 'nordpool') <state sensor.house_power_consumption=623; state_class=measurement, unit_of_measurement=W, device_class=power, icon=mdi:flash-outline, friendly_name=House - Power Consumption @ 2022-11-04T23:35:16.446685+01:00>|<state sensor.house_power_consumption=624; state_class=measurement, unit_of_measurement=W, device_class=power, icon=mdi:flash-outline, friendly_name=House - Power Consumption @ 2022-11-04T23:35:17.020556+01:00>
Unable to handle data: sensor.house_power_consumption ('HomeAssistantHub' object has no attribute 'nordpool') <state sensor.house_power_consumption=624; state_class=measurement, unit_of_measurement=W, device_class=power, icon=mdi:flash-outline, friendly_name=House - Power Consumption @ 2022-11-04T23:35:17.020556+01:00>|<state sensor.house_power_consumption=626; state_class=measurement, unit_of_measurement=W, device_class=power, icon=mdi:flash-outline, friendly_name=House - Power Consumption @ 2022-11-04T23:35:17.589973+01:00>
Unable to handle data: sensor.house_power_consumption ('HomeAssistantHub' object has no attribute 'nordpool') <state sensor.house_power_consumption=626; state_class=measurement, unit_of_measurement=W, device_class=power, icon=mdi:flash-outline, friendly_name=House - Power Consumption @ 2022-11-04T23:35:17.589973+01:00>|<state sensor.house_power_consumption=624; state_class=measurement, unit_of_measurement=W, device_class=power, icon=mdi:flash-outline, friendly_name=House - Power Consumption @ 2022-11-04T23:35:18.193698+01:00>
Unable to handle data: sensor.house_power_consumption ('HomeAssistantHub' object has no attribute 'nordpool') <state sensor.house_power_consumption=624; state_class=measurement, unit_of_measurement=W, device_class=power, icon=mdi:flash-outline, friendly_name=House - Power Consumption @ 2022-11-04T23:35:18.193698+01:00>|<state sensor.house_power_consumption=625; state_class=measurement, unit_of_measurement=W, device_class=power, icon=mdi:flash-outline, friendly_name=House - Power Consumption @ 2022-11-04T23:35:18.749007+01:00>
```

**Reply by elden1337** [reply]
Ok. I know what it is. I have accidentally crashed the setup for those not using nordpool pricing.

**Reply by elden1337** [reply]
https://github.com/elden1337/hass-peaq/issues/173

**Reply by elden1337** [reply]
@kirilldem can you try and update to this beta-release and see if it works for you? https://github.com/elden1337/hass-peaq/releases/tag/v2.2.0-beta

Include pre-releases in hacs to see it in the list.

**Comment by elden1337** [comment]
Im gonna convert this issue into a discussion instead. I realize there is need for a better guide on getting started now when more users are here.

**Reply by kirilldem** [reply]
Sorry, I didn't realise one could create a discussion. Thanks for converting it.

**Comment by elden1337** [comment]
i've started a wiki. do you have any thoughts on this? 
and have you gotten yours to work now?

https://github.com/elden1337/hass-peaq/wiki/Getting-started
